### PR TITLE
Add more complete 'turn off' documentation

### LINKED ISSUE #225: Include provider-general kubernetes teardown instructions
I just got dinged with a $150 charge from Amazon because I had mistakenly been running 150 volumes on EC2 that were created for users. If I would have followed the kubernetes teardown instructions instead of the "amazon button clicking" instructions this would have been avoided. We need something like

`kubectl delete ns <namespace>`

at the beginning of the tear-down page, and follow that with more provider-specific instructions if needed. (@yuvipanda can you confirm this is right?)

### PR DESCRIPTION
**Pull request by yuvipanda**
Fixes #225

### CODE REVIEW

File: doc/source/turn-off.rst [hunk 1]
```diff
@@ -7,18 +7,34 @@ When you are done with your hub, you should delete it so you are no longer
...
+
+1. First, delete the helm release. This deletes all resources that were created
+   by helm to make your jupyterhub.
+
+  .. code-block:: bash
+
+     helm delete <your-helm-release-name> --purge
 
-1. First, delete the namespace the hub was installed in. This deletes any disks
+2. First, delete the namespace the hub was installed in. This deletes any disks
```
Comment (willingc) [inline doc/source/turn-off.rst]: ~~First~~ Next

File: doc/source/turn-off.rst [hunk 2]
```diff
@@ -7,18 +7,34 @@ When you are done with your hub, you should delete it so you are no longer
...
    that may have been created to store user's data, and any IP addresses that
    may have been provisioned.
 
    .. code-block:: bash
 
       kubectl delete namespace <your-namespace>
 
-2. Next, you should delete the kubernetes cluster. You can list all the clusters
+Google Cloud Engine
+-------------------
```
Comment (willingc) [inline doc/source/turn-off.rst]: Let's drop the provider specific instructions one header level `~~~~~~~~~`

File: doc/source/turn-off.rst [hunk 3]
```diff
@@ -48,6 +64,9 @@ Google Cloud Engine
 Amazon AWS
 ----------
```
Comment (willingc) [inline doc/source/turn-off.rst]: Let's drop the provider specific instructions one header level ~~~~~~~~~

File: doc/source/turn-off.rst [hunk 4]
```diff
@@ -7,18 +7,34 @@ When you are done with your hub, you should delete it so you are no longer
...
 
    .. code-block:: bash
 
       kubectl delete namespace <your-namespace>
 
-2. Next, you should delete the kubernetes cluster. You can list all the clusters
+Google Cloud Engine
+-------------------
+
+Make sure you have performed the cloud provider agnostic steps listed above before
```
Comment (willingc) [inline doc/source/turn-off.rst]: Make this line an actual step. Maybe:

1. Perform the steps above for all cloud providers. These cloud provider agnostic steps will delete the helm chart and delete the hub's namespace. This must be done before proceeding.

File: doc/source/turn-off.rst [hunk 5]
```diff
@@ -31,7 +47,7 @@ Google Cloud Engine
 
       gcloud container clusters delete <CLUSTER-NAME> --zone=<CLUSTER-ZONE>
 
-3. Double check to make sure all the resources are now deleted, since anything you
+2. Double check to make sure all the resources are now deleted, since anything you
    have not deleted will cost you money! You can check the `web console <https://console.cloud.google.com>`_
    (make sure you are in the right project and account!) to make sure everything
```
Comment (willingc) [inline doc/source/turn-off.rst]: remove ! in this line

Comment (willingc) [inline doc/source/turn-off.rst]: ~~make sure~~ verify that everything

File: doc/source/turn-off.rst [hunk 6]
```diff
@@ -48,6 +64,9 @@ Google Cloud Engine
 Amazon AWS
 ----------
 
+Make sure you have performed the cloud provider agnostic steps listed above before
```
Comment (willingc) [inline doc/source/turn-off.rst]: Make this a step 1 like in the Google cloud section.

File: doc/source/turn-off.rst [hunk 7]
```diff
@@ -48,6 +64,9 @@ Google Cloud Engine
...
+doing this!
+
 The easiest way to delete your cloud resources on AWS is to use their
```
Comment (willingc) [inline doc/source/turn-off.rst]: Make this Step 2.

File: doc/source/turn-off.rst [hunk 8]
```diff
@@ -7,18 +7,34 @@ When you are done with your hub, you should delete it so you are no longer
 paying money for it. The following sections describe how to delete your
 JupyterHub resources on various cloud providers.
 
-Google Cloud Engine
--------------------
+For all cloud providers
+-----------------------
+
+The steps in this section must be performed for all cloud providers first,
+before doing the cloud provider specific setup.
```
Comment (choldgraf) [inline doc/source/turn-off.rst]: cloud provider-specific setup

File: doc/source/turn-off.rst (hunk 2 above)
Comment (choldgraf) [reply doc/source/turn-off.rst]: agree

File: doc/source/turn-off.rst [hunk 9]
```diff
@@ -7,18 +7,34 @@ When you are done with your hub, you should delete it so you are no longer
 paying money for it. The following sections describe how to delete your
 JupyterHub resources on various cloud providers.
```
Comment (choldgraf) [inline doc/source/turn-off.rst]: Could we add one sentence that describes a short high-level overview of what we'll do in the page? E.g.:

```
Tearing down your JupyterHub entails:

1. Deleting your Kubernetes namespace, which deletes all objects created and managed by Kubernetes
2. Deleting any computational resources you've requested from the cloud provider
3. Running a final check to make sure there aren't any lingering resources that haven't been deleted (e.g., storage volumes in some cloud providers)
```

**Comment by choldgraf** [comment]
A few comments - I'm happy w/ it once @willingc 's comments are addressed

File: doc/source/turn-off.rst (hunk 9 above)
Comment (willingc) [reply doc/source/turn-off.rst]: Great suggestion @choldgraf. This would be helpful at the start of the section 👍

**Comment by yuvipanda** [comment]
@choldgraf @willingc I updated!

**Comment by yuvipanda** [comment]
thanks for the writing feedback, @willingc and @choldgraf! Hopefully over time my writing skills will get better too :)

**Comment by willingc** [comment]
Your writing is awesome. Keep it flowing 👍

**Pull request merged**